workflow/templateresolution/context.go

```diff
@@ -16,6 +18,12 @@ import (
 // maxResolveDepth is the limit of template reference resolution.
 const maxResolveDepth int = 10
 
+// WorkflowTemplateNamespacedGetter abstracts over ways to resolve WorkflowTemplates.
+type WorkflowTemplateNamespacedGetter interface {
```

Comment:
Can you see if we can merge both `WorkflowTemplateNamespacedGetter` and `ClusterWorkflowTemplateGetter` into one interface/struct? I'm not sure why it needs to be both.

Reply:
Actually, working on a PoC for this... give me a bit and I'll show you what I mean

Reply:
I found a reason as to why we need to have them separate... ignore this for now.